web/libs/editor/src/components/Comments/Comment/CommentItem.tsx

```diff
@@ -42,198 +45,227 @@ interface CommentItemProps {
+              {`${isEdited ? "updated" : ""} ${humanDateDiff(time)}`}
+            </Tooltip>
           </Elem>
-          <TimeTracker />
-        </Space>
-      </Space>
-
-      <Elem name="content">
-        <Elem name="text">
-          {isEditMode ? (
-            <CommentFormBase
-              value={text}
-              onSubmit={async (value) => {
-                await updateComment(value);
-                setText(value);
-                await listComments({ suppressClearComments: true });
-              }}
-            />
-          ) : isConfirmDelete ? (
-            <Elem name="confirmForm">
-              <Elem name="question">Are you sure?</Elem>
-              <Elem name="controls">
-                <Button onClick={() => deleteComment()} size="compact" look="danger" autoFocus>
-                  Yes
-                </Button>
-                <Button onClick={() => setConfirmMode(false)} size="compact">
-                  No
-                </Button>
-              </Elem>
+        );
+      return null;
+    };
+
+    return (
+      <Block
+        name="comment-item"
+        mod={{ resolved, highlighted: isHighlighted }}
+        onMouseEnter={() => {
+          setHighlighted(true);
+        }}
+        onMouseLeave={() => {
+          setHighlighted(false);
+        }}
+        ref={_commentRef}
+      >
+        <Space spread size="medium" truncated>
+          <Space size="small" truncated>
+            <Elem tag={Userpic} user={createdBy} name="userpic" showUsername username={createdBy} />
+            <Elem name="name" tag="span">
+              {userDisplayName(createdBy)}
             </Elem>
-          ) : (
-            <>
-              {classifications?.default?.values?.length > 0 && (
-                <Elem name="classifications" tag="ul">
-                  {classifications?.default?.values?.map((valueArray: string[], index: number) => (
-                    <li key={index}>{valueArray.join("/")}</li>
-                  ))}
-                </Elem>
-              )}
-              {text}
-              {hasLinkState && (
-                <Elem name="linkState">
-                  <LinkState linking={linking} region={region} result={result} interactive />
+          </Space>
+
+          <Space size="small">
+            <Elem name="resolved" component={IconCheck} />
+            <Elem name="saving" mod={{ hide: isPersisted }}>
+              <Elem name="dot" />
+            </Elem>
+            <TimeTracker />
+          </Space>
+        </Space>
+
+        <Elem name="content">
+          <Elem name="text">
+            {isEditMode ? (
+              <>
+                <CommentFormBase
+                  value={text}
+                  onSubmit={async (value) => {
+                    await updateComment(value, classifications);
+                    setText(value);
+                    await listComments({ suppressClearComments: true });
+                  }}
```

Comment:
This should similarly to the other suggestions be extracted and wrapped to a useCallback with a dependency on classifications. Not sure what underpins the CommentFormBase but likely not as big of a problem for re-renders as Taxonomy could be.

Reply:
That is a great question. Only if the functions are useCallback based and require their own dependencies. If they are useCallback with no dependencies that is considered stable and not required. In general the best rule of thumb is add all dependencies except for refs (useRef() created values), externally defined functions. Now the only question that remains is to ensure the functions that are added as dependencies are properly using useCallback.

Reply:
These look like props, would definitely affect the render of this component if not properly defined with useCallback, but should be included in this useCallback based on the fact they are the prop based.